### PR TITLE
Introduce a new abstraction layer over island input/output data

### PR DESCRIPTION
**Pull request by rgarnov**
- Port GExecutor to this new interface

```
build_gapi_standalone:Linux x64=ade-0.1.1f
build_gapi_standalone:Win64=ade-0.1.1f
build_gapi_standalone:Mac=ade-0.1.1f
build_gapi_standalone:Linux x64 Debug=ade-0.1.1f
```

### Pull Request Readiness Checklist

See details at https://github.com/opencv/opencv/wiki/How_to_contribute#making-a-good-pull-request

- [x] I agree to contribute to the project under OpenCV (BSD) License.
- [x] To the best of my knowledge, the proposed patch is not based on a code under GPL or other license that is incompatible with OpenCV
- [x] The PR is proposed to proper branch
- [x] There is reference to original bug report and related work
- [x] There is accuracy test, performance test and test data in opencv_extra repository, if applicable
      Patch to opencv_extra has the same branch name.
- [x] The feature is well documented and sample code can be built with the project CMake

### CODE REVIEW

**Comment by dmatveev** [comment]
I believe it really waits for other MRs to be merged first (e.g. on fixing the Opaque support).

...Or not?

File: modules/gapi/src/executor/gexecutor.cpp [hunk 1]
```diff
@@ -117,11 +117,36 @@ void cv::gimpl::GExecutor::initResource(const ade::NodeHandle &orig_nh)
         // Constructed on Reset, do nothing here
         break;
 
+    // FIXME: What about GOPAQUE here??
```
Comment (dmatveev) [inline modules/gapi/src/executor/gexecutor.cpp:120]: ? :)

Comment (rgarnov) [reply modules/gapi/src/executor/gexecutor.cpp]: Added in https://github.com/opencv/opencv/pull/16964, removed this FIXME

File: modules/gapi/src/backends/fluid/gfluidbackend.hpp [hunk 2]
```diff
@@ -153,6 +153,8 @@ class GFluidExecutable final: public GIslandExecutable
     virtual void run(std::vector<InObj>  &&input_objs,
                      std::vector<OutObj> &&output_objs) override;
 
+    using GIslandExecutable::run; // (IInput&, IOutput&) version
```
Comment (dmatveev) [inline modules/gapi/src/backends/fluid/gfluidbackend.hpp:156]: Is this required?

Comment (rgarnov) [reply modules/gapi/src/backends/fluid/gfluidbackend.hpp:156]: Yes, there were warnings on android and mac that another (non virtial) run below hides this method

**Comment by rgarnov** [comment]
@alalek could you merge this please?

File: modules/gapi/src/backends/fluid/gfluidbackend.hpp (hunk 2 above)
Comment (dmatveev) [reply modules/gapi/src/backends/fluid/gfluidbackend.hpp:156]: ok I see now. Thanks!

**Pull request merged**